atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/TableDefinition.java

```diff
@@ -115,6 +121,9 @@ public void rowComponent(String componentName, ValueType valueType) {
 
     public void rowComponent(String componentName, ValueType valueType, ValueByteOrder valueByteOrder) {
         Preconditions.checkState(state == State.DEFINING_ROW_NAME);
+        if (rowNameComponents.isEmpty() && CRITICAL_ROW_TYPES.contains(valueType)) {
+            log.error("First row component {} of type {} is an antipattern.", componentName, valueType);
```

Comment:
This should probably have more of an explanation of why it's bad and what good alternatives are.